Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewColumnCollection.cs

```diff
@@ -0,0 +1,287 @@
+﻿// CodeContracts
+// 
+// Copyright (c) Microsoft Corporation
+// 
+// All rights reserved. 
+// 
+// MIT License
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections;
+using System.ComponentModel;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Contracts;
+
+namespace System.Windows.Forms
+{
+    /// <summary>
+    /// Represents a collection of <see cref="T:System.Windows.Forms.DataGridViewColumn"/> objects in a <see cref="T:System.Windows.Forms.DataGridView"/> control.
+    /// </summary>
+    public class DataGridViewColumnCollection : BaseCollection // IList, 
+    {
+        // <summary>
+        // Gets the <see cref="T:System.Windows.Forms.DataGridView"/> upon which the collection performs column-related operations.
+        // </summary>
+        // 
+        // <returns>
+        // <see cref="T:System.Windows.Forms.DataGridView"/>.
+        // </returns>
+        // protected DataGridView DataGridView { get; }
+        
+        /// <summary>
+        /// Gets or sets the column at the given index in the collection.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The <see cref="T:System.Windows.Forms.DataGridViewColumn"/> at the given index.
+        /// </returns>
+        /// <param name="index">The zero-based index of the column to get or set.</param><exception cref="T:System.ArgumentOutOfRangeException"><paramref name="index"/> is less than zero or greater than the number of columns in the collection minus one.</exception>
+        public DataGridViewColumn this[int index]
+        {
+            get
+            {
+                Contract.Requires(index >= 0);
```

Comment:
I think in most other cases we used one requires with two checks instead of one..

I.e. should be `index >= 0 && index < Count`.